src/payment/strategies/afterpay/afterpay-payment-strategy.ts

```diff
@@ -125,4 +136,24 @@ export default class AfterpayPaymentStrategy implements PaymentStrategy {
             return 'AU';
         }
     }
+
+    private _outsideLimits(amount: number, paymentMethod?: PaymentMethod): boolean {
+        const $minimum = paymentMethod?.initializationData?.min || 0;
+
+        return amount < $minimum || amount > paymentMethod?.initializationData?.max;
+    }
+
+    private async _loadPaymentMethod(gatewayId: string, methodId: string, options?: RequestOptions): Promise<InternalCheckoutSelectors> {
```

Comment:
When purchasing outside the Afterpay limits, bigpay will not return Afterpay while fetching the paymentMethods, we want to catch that error (404 Not Found) and replace it with a custom message, any other error will be re threw.

Reply:
I'm probably missing some info, but how is it possible to reach this scenario?

Does the `payments` endpoint method not filter out Afterpay if the payable value is too big/small?
If it did, this strategy would never be initialised and wouldn't need to handle this eventuality.

Reply:
The problem occurs when the shopper adds a coupon/gift certificate in the last step of the checkout, at this point the payments endpoint returned Afterpay as payment method because the payable value is within the limits, but when the shopper adds the coupon that new payable amount may not be within the limits and when we try to get the initialization token (necessary to initialize their SDK) the payments endpoint will return an Not Found error because now the payment method does not meet the requirements around the order amount and it is not supported as payment method

Reply:
@leeBigCommerce The problem occurs when the shopper adds a coupon/gift certificate in the last step of the checkout, at this point the payments endpoint returned Afterpay as payment method because the payable value is within the limits, but when the shopper adds the coupon that new payable amount may not be within the limits and when we try to get the initialization token (necessary to initialize their SDK) the payments endpoint will return an Not Found error because now the payment method does not meet the requirements around the order amount and it is not supported as payment method

Reply:
Sounds like we should be refetching available payment method options whenever the payable amount changes (including coupons and gift certificates), but I'm guessing this would be a bigger change to plan in @davidchin ?

If that is too big for now, can we change the endpoint to return a `400` or `422` in this scenario (perhaps with a more specific error key in the body)?

Currently this is a pretty big stretch of what `404` is meant to represent

Reply:
@leeBigCommerce We think it's a good idea to change the returned status to 422 (it will be easier to filter the error code), but we believe that this change has to be part of a new ticket since all the providers use this endpoint, we'll first need to analyze what could be the impacted areas around all the platform and then run regression tests in several providers to ensure functionality is preserved across the platform, for this reason, we suggest to extract these changes you are proposing to a new ticket.

Reply:
Sounds good 👍 

Maybe add some inline comments around the 404 uses to explain the reason and temporary nature to developers?

This test in particular makes it seem like desired behaviour: https://github.com/bigcommerce/checkout-sdk-js/pull/1175/files#diff-9dc05fcb1acb1dc5a9246aaa802472601f28f2eab3516bd09d0e1cb856ac605cR202 - not mentioning that the 404 implicitly represents a 'invalid amount' issue.

Reply:
@leeBigCommerce I've been looking at the error code and where and why is raised as 404, is raised in BCApp whenever the payment provider is present in BCApp but not in BigPay, I will depend on the provider and why it can't be initialized in this step of the checkout process, for Afterpay it's because of the amount, but another provider could fail for other reason.

I believe tthat the 404 is the correct error code, the provider was not found, and the strategy should explain why.

Reply:
the Payments team is going to review the use of `404` meaning both 'not found' and 'unavailable' outside of this ticket as it seems to be a bit of a wider issue